src/auth/src/credentials.rs

```diff
@@ -201,6 +201,41 @@ pub(crate) mod dynamic {
     }
 }
 
+/// Represents the type of credential as specified in the credentials file.
+#[derive(Debug, PartialEq)]
+pub enum CredentialType {
+    /// User account credentials, typically from gcloud auth
+    AuthorizedUser,
+    /// Service account key file credentials
+    ServiceAccount,
+    /// External account credentials (e.g., Workload Identity Federation)
+    ExternalAccount,
+}
+
+impl CredentialType {
+    fn from_str(type_str: &str) -> Result<Self> {
+        match type_str {
```

Comment:
comment: We have a credential type (MDS credential) that is NOT in the ADC json. So there is no string for it.

That is fine, just an FYI.